src/platform.ts

```diff
@@ -247,6 +256,22 @@ export function getRuntimeDisplayName(runtime: Runtime): string {
     }
 }
 
+/**
+ * Get the fallback runtime.
+ */
+export function getFallbackRuntime(runtime: Runtime): Runtime {
+    switch (runtime) {
+        case Runtime.Ubuntu_22:
+        case Runtime.Ubuntu_20:
+        case Runtime.Ubuntu_18:
+            return Runtime.Ubuntu_16;

```

Comment:
Add a comment about why 16. 

But also it feels like the intent here is basically to get the max supported version - for example right now if the platform is 22 but the package only contains 18 then this will fail (or get 16 if it also includes 16). So if we're going to be doing this kind of thing then it would probably be best to update the logic to grab the "closest" one it can (version or architecture). 

I do wonder about the validity of doing this though - we're basically assuming that someone working for 16 will also work for 22, which as the MySQL people have found out isn't necessarily the case. We may instead want to throw an error about an unsupported platform so the user knows exactly what the problem is and the package owner can build and verify the new package themselves before releasing it to avoid any unintended behavior/issues.

Reply:
now I have the full fallback logic added.